azext_edge/edge/params.py

```diff
@@ -302,6 +310,822 @@ def load_iotops_arguments(self, _):
+            "Operations deployment might result in data loss.",
+        )
+        context.argument(
+            "keep_alive",
+            options_list=["--keep-alive", "--ka"],
+            help="The maximum time in seconds that the data flow client can "
+            "be idle before sending a PINGREQ message to the broker. "
+            "Min value: 0.",
+            type=int,
+        )
+        context.argument(
+            "max_inflight_messages",
+            options_list=["--max-inflight-msg", "--mim"],
+            help="the maximum number of inflight messages that the data "
+            "flow MQTT client can have. Min value: 0.",
+            type=int,
+        )
+        context.argument(
+            "protocol",
+            options_list=["--protocol"],
+            arg_type=get_enum_type(ListenerProtocol, default=ListenerProtocol.MQTT.value),
+            help="Enable or disable websockets.",
+        )
+        context.argument(
+            "qos",
+            options_list=["--qos"],
+            help="Quality of Service (QoS) level for the MQTT messages.",
+            type=int,
+        )
+        context.argument(
+            "retain",
+            options_list=["--retain"],
+            arg_type=get_enum_type(MqttRetainType, default=MqttRetainType.KEEP.value),
+            help="Retain setting to specify whether the data flow should keep the retain "
+            "flag on MQTT messages. Setting this ensures whether or not the remote broker "
+            "has the same messages retained as the local broker.",
+        )
+        context.argument(
+            "session_expiry",
+            options_list=["--session-expiry"],
+            help="The session expiry interval in seconds for the data flow MQTT client. "
+            "Min value: 0.",
+            type=int,
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create adx") as context:
+        context.argument(
+            "database_name",
+            options_list=["--database-name", "--db"],
+            help="The name of the Azure Data Explorer database.",
+        )
+        context.argument(
+            "host",
+            options_list=["--host"],
+            help="Host of the Azure Data Explorer is "
+            "Azure Data Explorer cluster URI. In the form "
+            "of https://cluster.region.kusto.windows.net",
+        )
+        context.argument(
+            "authentication_type",
+            options_list=["--auth-type"],
+            choices=CaseInsensitiveList(
+                [
+                    DataflowEndpointAuthenticationType.SYSTEMASSIGNED.value,
+                    DataflowEndpointAuthenticationType.USERASSIGNED.value,
+                ]
+            ),
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create adls") as context:
+        context.argument(
+            "secret_name",
+            options_list=["--secret-name", "--sn"],
```

Comment:
nit: wanna reopen the discourse around --secret vs --secret-name?

I think secret-name is fine here but storage-account-name needs to have no -name

Reply:
--secret-name is good, made same comment about the value of -name for --storage-account

Reply:
It would be good to align on the short alias for `--secret-name`, here it is `--sn` but for `az iot ops connector opcua *` commands it is `-s`.

Reply:
lets do --secret-name and -s for now